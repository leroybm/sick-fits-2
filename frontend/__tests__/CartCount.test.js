import { render } from '@testing-library/react';
import { waitFor } from '@babel/core/lib/gensync-utils/async';
import CartCount from '../components/CartCount';

describe('<CartCount />', () => {
  it('should render', () => {
    render(<CartCount count={10} />);
  });

  it('should match snapshot', () => {
    const { container } = render(<CartCount count={11} />);

    expect(container).toMatchSnapshot();
  });

  it('should update via props', () => {
    const { container, rerender, debug } = render(<CartCount count={12} />);

    expect(container.textContent).toBe('12');
    rerender(<CartCount count={13} />);
    expect(container.textContent).toBe('1312');
    waitFor(() => {
      expect(container.textContent).toBe('13');
      expect(container).toMatchSnapshot();
    });
  });
});
