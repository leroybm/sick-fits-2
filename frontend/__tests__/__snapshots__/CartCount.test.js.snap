// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CartCount /> should match snapshot 1`] = `
<div>
  <span
    class="CartCount__AnimationStyles-sc-xxvp4g-1 gwXiGr"
  >
    <div>
      <div
        class="CartCount__Dot-sc-xxvp4g-0 ekXcLS count"
      >
        11
      </div>
    </div>
  </span>
</div>
`;
